src/Components/Components/src/Auth/AuthenticationStateProvider.razor

```diff
@@ -0,0 +1,43 @@
+@namespace Microsoft.AspNetCore.Components
+@implements IDisposable
+
+<CascadingValue T="IAuthenticationState" Value="@_currentAuthenticationState" ChildContent="@ChildContent" />
+
+@functions {
+    private IAuthenticationState _currentAuthenticationState;
+
+    [Inject] private IAuthenticationStateProvider AuthStateProviderService { get; set; }
```

Comment:
nit: ProviderService makes me sad. It's like UtilityHelper. I would either pick Provider or Service

Reply:
OK, this name isn't part of the public API surface but I'm happy to change it.